stats/stats.go

```diff
@@ -20,13 +29,15 @@ type key struct {
 }
 
 type stats struct {
-	allMetrics map[key]prom.Counter
+	counters map[key]prom.Counter
+	gauges   map[key]prom.Gauge
 }
 
 const (
 	subsystemName = "stats"
 
 	unknownCounter   = "unknown_counter"
```

Comment:
the stats package isn't critical for file.d, it works as an auxiliary module.  
that way, in case a metric wasn't initialised, the main system (file.d i/o & actions) shouldn't fail. that's why we added this "unknown" metric that basically gets incremented in case we try to increment an unregistered metric.

@vano144 @ansakharov what's your opinion on this?

Reply:
if a metric is not initialized at start and is requested, this is a bug 
current solution, on opposite, is a road to more subtle bugs

Reply:
I see three ways:
- Return error: user (plugin in file.d) should decide how handle this err. But why should it decide? Not it's responsibility.
- Register metric on demand. But why it wasn't registered before? Something went wrong.
- Log this situation, probably increment other critical metric and fail fast.

@andrewmed suggested fail fast and it seems reasonable to me. Unregistered metric is fiasco. 
After all you can easily test if it registered: write simple test where you creating plugin and checking it's metric in `stats` pkg.

Neither you not I want spend time searching for phantom metric.

You probably would like take a look about `fail fast` conception.
For example: https://www.yegor256.com/2015/08/25/fail-fast.html

Reply:
* You will not look at the logs when the application starts, that's why the metric is needed
* I don't think it's a reason to fall

Reply:
We decided to fix this issue later in a separate issue. 
cc @andrewmed @ansakharov

Reply:
@vano144 it's reason to fail if we have proper tests, I'll suggest PR tomorrow.